tests/04_api/test_dataset_api.py

```diff
@@ -260,12 +260,6 @@ async def test_delete_dataset_cascade_jobs(
             status=JobStatusType.SUBMITTED,  # this is why ds are not deletable
             **common_args,
         )
-        await job_factory(
-            input_dataset_id=ds_not_deletable_2.id,
-            output_dataset_id=ds_not_deletable_1.id,
```

Comment:
If we remove this, why is `ds_non_deletable_1` non deletable?

Reply:
Note that there is probably a small bug a few lines below, as we are making the same call twice:
```python
        res = await client.delete(
            f"api/v1/project/{project.id}/dataset/{ds_not_deletable_1.id}/"
        )
        assert res.status_code == 422
        res = await client.delete(
            f"api/v1/project/{project.id}/dataset/{ds_not_deletable_1.id}/"
        )
        assert res.status_code == 422
```

Reply:
> If we remove this, why is `ds_non_deletable_1` non deletable?

because of this `fractal_server/app/routes/api/v1/dataset.py:
```python
# Fail if there exist jobs that are submitted and in relation with the 
# current dataset.
stm = _get_submitted_jobs_statement().where(
    or_(
        ApplyWorkflow.input_dataset_id == dataset_id,
        ApplyWorkflow.output_dataset_id == dataset_id,
    )
)
res = await db.execute(stm)
jobs = res.scalars().all()
if jobs:
    string_ids = str([job.id for job in jobs])[1:-1]
    raise HTTPException(
        status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
        detail=(
            f"Cannot delete dataset {dataset.id} because it "
            f"is linked to active job(s) {string_ids}."
        ),
    )
```
and
```python
await job_factory(
    input_dataset_id=ds_not_deletable_1.id,
    output_dataset_id=ds_not_deletable_2.id,
    status=JobStatusType.SUBMITTED,  # this is why ds are not deletable
    **common_args,
)
```
so both `ds_not_deletable_1` and `ds_not_deletable_2` can't be deleted

Reply:
Thanks for clarifying!

(my mistake is that was rather thinking about the case where we prevent two jobs from running with the same _output_ dataset, and in that case the _input_ dataset is not relevant).